woocommerce/woocommerce-gutenberg-products-block
# Create AssetsController and BlockTypesController (replaces Assets.php and Library.php)

Refactors Assets and Library with dependency injection. Closes #3875 and fixes #3868

I've marked this as in need of a dev note since we're removing classes, however, they are internal so unlikely to be used directly. cc @nerrad 

Will add inline notes for other changes.

### How to test the changes in this Pull Request:

General smoke test to ensure classes are still loaded. Check all block types are registered and existing blocks load within admin and frontend.

### Dev Note

> Two internal classes (`Automattic\WooCommerce\Blocks\Assets` and `Automattic\WooCommerce\Blocks\Library`) have been deprecated in this release, with functionality being moved to `AssetsController` and `BlockTypesController`. Since these were internal bootstrapping classes we don't anticipate any external usage, but noting that these classes will be removed in a future release of Blocks.

### Changelog

> Introduced AssetsController and BlockTypesController classes (which replace Assets.php and Library.php).
